test/malli/parser_test.cljc

```diff
@@ -0,0 +1,209 @@
+            [malli.core :as m]
+            [malli.edn :as edn]
+            [malli.generator :as mg]
+            [malli.error :as me]
+            [malli.impl.util :as miu]
+            [malli.registry :as mr]
+            [malli.transform :as mt]
+            [malli.util :as mu]
+            #?(:clj [malli.test-macros :refer [when-env]]))
+  #?(:clj  (:import (clojure.lang IFn PersistentArrayMap PersistentHashMap))
+     :cljs (:require-macros [malli.test-macros :refer [when-env]])))
+
+(defn simple-parser? [s] (boolean (:simple-parser (m/-parser-info (m/schema s)))))
+
+(def inheriting-parser-templates
+  "Schemas templates which have simple parsers iff ::HOLE has a simple parser.
+  Should also be generatable for any ::HOLE and be capable to (un)parsing
+  to a different value than its input if transforming."
+  [::HOLE
+   [:maybe ::HOLE]
+   [:schema ::HOLE]
+   [:schema {:registry {::a ::HOLE}} ::a]
+   [:schema {:registry {::a ::HOLE}} [:ref ::a]]
+   [:schema {:registry {::a [:ref ::b] ::b ::HOLE}} [:ref ::a]]
+   [:tuple ::HOLE]
+   [:tuple ::HOLE :any]
+   [:vector ::HOLE]
+   [:set ::HOLE]
+   [:seqable ::HOLE]
+   [:map [:foo ::HOLE]]
+   [:map [:foo {:optional true} ::HOLE]]
+   [:map [:foo ::HOLE] [:bar :int]]
+   [:and ::HOLE] ;; generator will fail if :any is first
+   [:and ::HOLE :any]
+   [:and ::HOLE :any :any]
+   [:or ::HOLE] ;; parser will always be identical if :any is first
+   [:or ::HOLE :any]
+   [:map-of ::HOLE :any]
+   [:map-of :any ::HOLE]
+   [:map-of ::HOLE ::HOLE]])
+
+(def simple-parser-templates
+  "Schema templates which have simple parsers for any value of ::HOLE."
+  [[:and {:parse 1} ::HOLE :any]
+   [:and {:parse :none} ::HOLE :any]
+   [:every ::HOLE]
+   [:-> ::HOLE]
+   [:function [:-> ::HOLE]]])
+
+(def transforming-parser-templates
+  "Schema templates which have transforming parsers for any value of ::HOLE."
+  [[:multi {:dispatch #'any?} [true ::HOLE]]
+   [:multi {:dispatch #'boolean} [true :any] [false ::HOLE]]
+   [:multi {:dispatch #'boolean} [true ::HOLE] [false :any]]
+   [:andn [0 ::HOLE]]
+   [:andn [0 ::HOLE] [1 :any]] ;; generator will fail if :any is first
+   [:orn [0 ::HOLE]]
+   [:orn [0 ::HOLE] [1 :any]]
+   [:orn [0 :any] [1 ::HOLE]]
+   [:orn [0 ::HOLE] [1 ::HOLE]]])
+
+(def simple-parser-schemas
+  "Schemas with simple parsers."
+  [:any
+   [:and :any]
+   :int
+   #'map?
+   :tuple
+   [:fn {:gen/schema :any} #'any?]
+   [:= 42] [:enum 42] [:not= 42] [:< 5] [:> 5] [:<= 5] [:>= 5]
+   #?@(:cljs [] :default [[:re #""]]) ;; no generator in cljs
+   :nil
+   :qualified-symbol
+   :uuid
+   [:not [:= (random-uuid)]] ;; generator is too unreliable to nest
+   :some])
+
+(def transforming-parser-schemas
+  "Schemas with transforming parsers."
+  [[:andn [:any :any]] [:catn [:any :any]] [:seqable [:catn [:any :any]]] [:multi {:dispatch #'any?} [true :any]]])
+
+(defn ensure-parser-type [expected-simple s]
+  #?(:bb nil ;;FIXME test.chuck incompatibility
```

Comment:
what's the problem?

Reply:
`malli.generator` doesn't work in bb yet.

Reply:
should the comment say exactly that, then?